apps/condo/domains/banking/schema/BankAccount.test.js

```diff
@@ -0,0 +1,305 @@
+                expect(bankAccount.tin).toBeDefined()
+                expect(bankAccount.country).toEqual('ru')
+                expect(bankAccount.routingNumber).toBeDefined()
+                expect(bankAccount.number).toBeDefined()
+                expect(bankAccount.currencyCode).toEqual('RUB')
+            })
+
+            test('user can\'t', async () => {
+                const admin = await makeLoggedInAdminClient()
+                const user = await makeClientWithNewRegisteredAndLoggedInUser()
+
+                const [ organization ] = await createTestOrganization(admin)
+
+                await expectToThrowAccessDeniedErrorToObj(async () => {
+                    await createTestBankAccount(user, organization)
+                })
+            })
+
+            test('anonymous can\'t', async () => {
+                const admin = await makeLoggedInAdminClient()
+                const anonymous = await makeClient()
+
+                const [ organization ] = await createTestOrganization(admin)
+
+                await expectToThrowAuthenticationErrorToObj(async () => {
+                    await createTestBankAccount(anonymous, organization)
+                })
+            })
+        })
+
+        describe('read', () => {
+            test('admin can', async () => {
+                const admin = await makeLoggedInAdminClient()
+                const [organization] = await createTestOrganization(admin)
+
+                const [createdObj] = await createTestBankAccount(admin, organization)
+                const [readObj] = await BankAccount.getAll(admin, { id: createdObj.id })
+
+                expect(createdObj.organization.id).toEqual(readObj.organization.id)
+                expect(createdObj.tin).toEqual(readObj.tin)
+                expect(createdObj.country).toEqual(readObj.country)
+                expect(createdObj.routingNumber).toEqual(readObj.routingNumber)
+                expect(createdObj.number).toEqual(readObj.number)
+                expect(createdObj.currencyCode).toEqual(readObj.currencyCode)
+            })
+
+            test('support can', async () => {
+                const admin = await makeLoggedInAdminClient()
+                const [organization] = await createTestOrganization(admin)
+
+                const support = await makeClientWithSupportUser()
+
+                const [createdObj] = await createTestBankAccount(admin, organization)
+                const [readObj] = await BankAccount.getAll(support, { id: createdObj.id })
+
+                expect(createdObj.organization.id).toEqual(readObj.organization.id)
+                expect(createdObj.tin).toEqual(readObj.tin)
+                expect(createdObj.country).toEqual(readObj.country)
+                expect(createdObj.routingNumber).toEqual(readObj.routingNumber)
+                expect(createdObj.number).toEqual(readObj.number)
+                expect(createdObj.currencyCode).toEqual(readObj.currencyCode)
+            })
+
```

Comment:
you also need a test for the case then the Account is already approved and the user trying to change tin/number or organization or some other field

Reply:
It is here. ApprovedAt and ApprovedBy are dropped to false if model changes